### PR TITLE
fix(grafana_user): add admin permission on new admin creation

### PR DESCRIPTION
**Pull request by pomverte**
##### SUMMARY
- Admin creation is bugged
- Workaround: you had to go through user update to become admin 🙄 

⚠️ beware, Grafana has 2 levels of admin role:
- at Server Admin level (managed by this collection)
- at Organization (not managed by this collection currently)

##### ISSUE TYPE
- 🐛 Bugfix Pull Request #142

##### COMPONENT NAME
grafana_user.py

##### ADDITIONAL INFORMATION
```
ansible-test integration grafana_user
```

### CODE REVIEW

File: plugins/modules/grafana_user.py [hunk 1]
```diff
@@ -272,11 +272,12 @@ def main():
     # search user by login
     target_user = grafana_iface.get_user_from_login(login)
```
Comment (pomverte) [inline plugins/modules/grafana_user.py:273]: `target_user` should be renamed to express that it is the user currently existing (or not) on grafana

File: tests/integration/targets/grafana_user/tasks/main.yml [hunk 2]
```diff
@@ -1,4 +1,43 @@
...
+    name: "Harley Quinn"
+    email: harley.quinn@gotham.city
+    login: harley
+    password: Wy3ta6ob6M3wHELv58MPfqOe126RTnWpcYfEhyJm
+    is_admin: true
+    state: present
+  register: result
+- assert:
+    that:
+      - "result.changed == false"
```
Comment (vnea) [inline tests/integration/targets/grafana_user/tasks/main.yml:36]: Am I wondering if we should also do this check : `- "result.failed == false"` ?

File: plugins/modules/grafana_user.py (hunk 1 above)
Comment (vnea) [reply plugins/modules/grafana_user.py:273]: Maybe we could name it `stored_user` or `stored_grafana_user` ?

File: tests/integration/targets/grafana_user/tasks/main.yml (hunk 2 above)
Comment (pomverte) [reply tests/integration/targets/grafana_user/tasks/main.yml:36]: No need: if `result.failed` the the previous task would stop the play

File: plugins/modules/grafana_user.py (hunk 1 above)
Comment (pomverte) [reply plugins/modules/grafana_user.py:273]: `actual_grafana_user` ?

Comment (vnea) [reply plugins/modules/grafana_user.py:273]: Seems OK for me.

File: plugins/modules/grafana_user.py [hunk 3]
```diff
@@ -286,7 +287,7 @@ def main():
             user = grafana_iface.update_user(target_user_id, email, name, login)
             module.exit_json(changed=True, user=user)
```
Comment (seuf) [inline plugins/modules/grafana_user.py:288]: I suggest to to same as above :
```
- user = grafana_iface.update_user(target_user_id, email, name, login)
- module.exit_json(changed=True, user=user)
+ target_user = grafana_iface.update_user(target_user_id, email, name, login)
+ has_changed = True

```

Comment (pomverte) [reply plugins/modules/grafana_user.py]: Done
Thanks for the review !

File: plugins/modules/grafana_user.py [hunk 4]
```diff
@@ -270,28 +270,29 @@ def main():
...
+    actual_grafana_user = grafana_iface.get_user_from_login(login)
     if state == 'present':
+        has_changed = False
 
-        if target_user is None:
+        if actual_grafana_user is None:
             # create new user
-            user = grafana_iface.create_user(name, email, login, password)
-            module.exit_json(changed=True, user=user)
+            actual_grafana_user = grafana_iface.create_user(name, email, login, password)
```
Comment (seuf) [inline plugins/modules/grafana_user.py:279]: maybe add a check here to test if the new created user is admin ?

Comment (rrey) [reply plugins/modules/grafana_user.py:279]: it's right after line 285

**Review by rrey — APPROVED** [review]
LGTM

**Comment by rrey** [comment]
🚀

**Comment by rrey** [comment]
@all-contributors add  @vnea for code

**Comment by rrey** [comment]
@all-contributors add @rrey for code

(about time :P )

File: plugins/modules/grafana_user.py (hunk 4 above)
Comment (pomverte) [reply plugins/modules/grafana_user.py:279]: at this particular line, the test will always be negative: user creation doesn't allow setting user's permissions

this is done [here](https://github.com/ansible-collections/community.grafana/pull/154/files#diff-3fc8f0dba81b075c6a8ec8b4b3eb63fb97287f9301cfcc599aa0150abd0e8ce9R286)

**Pull request merged**